portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,81 @@
 
 package com.google.sps.servlets;
 
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 import java.io.IOException;
+import java.util.ArrayList;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Returns an array of comments in JSON form */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+  int maxComments=3;
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  /* Returns an array of comments in JSON 
+   * Size of array is maxComments */
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+  public void doGet(final HttpServletRequest request, final HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+    ArrayList<String[]> comments = new ArrayList<String[]>();
+    int index=0;
+    for (Entity entity : results.asIterable()) {
+      if(index<maxComments){
```

Comment:
I think this code is a bit hard to parse at first since it's unclear what's a "Comment" versus an "Entity". It seems like what you're doing is essentially getting the first three entities and transforming them into comments? (But let me know if I'm misinterpreting.) So maybe instead of this, you could just slice the results from 0 to MAX_COMMENTS, then iterate through the sliced version to transform them into comments?

As an additional note, it might help to comment this bit of code.

Feel free to push back on this if it doesn't make much sense.

Reply:
You're right, I moved this inside a new function and it now iterates only over a sliced version of the results. I think it makes more sense now.